crates/sparrow-runtime/src/streams/pulsar_stream.rs

```diff
@@ -193,26 +226,28 @@ impl PulsarReader {
             }
         }
 
-        // ensure that _publish_time never goes backwards
-        for avro_value in &mut avro_values {
-            for (field_name, field_value) in avro_value {
-                if field_name == "_publish_time" {
-                    match field_value {
-                        Value::TimestampMillis(publish_time) => {
-                            if *publish_time < self.last_publish_time {
-                                *publish_time = self.last_publish_time;
-                            } else {
-                                self.last_publish_time = *publish_time;
+        if self.require_ordered_publish_time {
```

Comment:
Just added this flag so I can reuse the `pulsar_stream` code without pulling a lot of the logic into a separate path.